PubNub/Core/PubNub+PAM.h

```diff
@@ -0,0 +1,49 @@
+#import "PubNub+Core.h"
+#import "PNPAMToken.h"
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+#pragma mark API group interface
+
+/**
+ * @brief \b PubNub client core class extension to provide access to 'PAM' API group.
+ *
+ * @discussion Set of API which allow to change access token used by client.
+ *
+ * @author Serhii Mamontov
+ * @version 4.17.0
+ * @since 4.17.0
+ * @copyright © 2010-2021 PubNub, Inc.
+ */
+@interface PubNub (PAM)
+
+
+#pragma mark - PAM
+
+/**
+ * @brief Decode an existing token and returns the object containing permissions embedded in that token. The client may use this method for debugging.
+ *
+ * @param token Base64-encoded PubNub access token.
+ *
+ * @return Decoded token representation instance.
+ */
+- (nullable PNPAMToken *)parseAuthToken:(NSString *)token;
+
+/**
+ * @brief Set PubNub access token which should be used to authorize REST API calls.
+ *
+ * @code
+ * [self.client setToken:@"access-token"];
+ * @endcode
+ *
+ * @param token Base64-encoded PubNub access token.
+ */
+- (void)setAuthToken:(NSString *)token;
```

Comment:
In Python and Java we used a "setToken/set_token" method name for this one. Maybe we should be consistent across all SDKs?

Reply:
@650elx i don’t know why we end up with so simplified method name. What if later we will have to add another type of token?

Reply:
It was already established name in Java SDK

Reply:
@parfeon I guess this is a question to @raipubnub  - we've used JS as a reference implementation

Reply:
@650elx I probably will remove this method and document way to change it in same way as it currently done for other client properties change: https://www.pubnub.com/docs/sdks/objective-c/api-reference/configuration#authentication-key

Reply:
Just want to add my support for keeping the specified method name.  Apple purposefully names properties/methods/objects to provide clarity into what their function is, so `setAuthToken` aligns with platform naming conventions.

Reply:
@CraigLn I've kept method after getting more info from @raipubnub :)